SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestAttachedComponentReplication/SpatialTestAttachedComponentReplication.cpp

```diff
@@ -0,0 +1,218 @@
+		break;
+	case ESpatialTestAttachedComponentReplicationType::DynamicallySpawnedWithDynamicComponent:
+		AddStep(TEXT("Spawn a test actor and attach dynamic component to it"), FWorkerDefinition::Server(1), nullptr, [this] {
+			AActor* ActorWithDynamicComponent = GetWorld()->SpawnActor<ASpatialTestAttachedComponentReplicationActorWithDynamicComponent>();
+			UActorComponent* DynamicComponent = NewObject<USpatialTestAttachedComponentReplicationComponent>(ActorWithDynamicComponent);
+			DynamicComponent->RegisterComponent();
+			RegisterAutoDestroyActor(ActorWithDynamicComponent);
+			FinishStep();
+		});
+		break;
+	case ESpatialTestAttachedComponentReplicationType::DynamicallySpawnedWithDefaultComponent:
+		AddStep(TEXT("Spawn a test actor with an attached default component"), FWorkerDefinition::Server(1), nullptr, [this] {
+			AActor* TestActorWithDefaultComponent =
+				GetWorld()->SpawnActor<ASpatialTestAttachedComponentReplicationActorWithDefaultComponent>();
+			RegisterAutoDestroyActor(TestActorWithDefaultComponent);
+			FinishStep();
+		});
+		break;
+	default:
+		checkNoEntry();
+	}
+
+	AddWaitingStep(DescribeTestType(TestType) + TEXT("Retrieve the level placed actor"), FWorkerDefinition::AllWorkers, [this, TestType] {
+		UClass* ActorClassToLookFor = GetClassForTestType(TestType);
+
+		TArray<AActor*> LevelPlacedActors;
+		UGameplayStatics::GetAllActorsOfClass(this, ActorClassToLookFor, LevelPlacedActors);
+
+		if (LevelPlacedActors.Num() > 1)
+		{
+			AddError(FString::Printf(TEXT("Received %d actors while only 1 expected"), LevelPlacedActors.Num()));
+		}
+		else if (LevelPlacedActors.Num() == 1)
+		{
+			LevelPlacedActor = Cast<ASpatialTestAttachedComponentReplicationActor>(LevelPlacedActors[0]);
+		}
+		RequireTrue(IsValid(LevelPlacedActor), TEXT("Discovered actor is valid"));
+
+		constexpr float ActorPollDuration = 3.0f;
+
+		RequireCompare_Float(TimeInStep, EComparisonMethod::Greater_Than, ActorPollDuration,
+							 FString::Printf(TEXT("Step ran for %f secs"), ActorPollDuration));
+	});
+
+	AddWaitingStep(DescribeTestType(TestType) + TEXT("Retrieve the attached component"), FWorkerDefinition::AllWorkers, [this] {
+		TArray<UActorComponent*> AttachedComponents;
+
+		Algo::CopyIf(LevelPlacedActor->GetComponents(), AttachedComponents, [](const UActorComponent* Component) {
+			return Component->IsA<USpatialTestAttachedComponentReplicationComponent>();
+		});
+
+		if (AttachedComponents.Num() > 1)
+		{
+			AddError(FString::Printf(TEXT("Received %d components while only 1 expected"), AttachedComponents.Num()));
+		}
+		else if (AttachedComponents.Num() == 1)
+		{
+			AttachedComponent = Cast<USpatialTestAttachedComponentReplicationComponent>(AttachedComponents[0]);
+		}
+		RequireTrue(IsValid(AttachedComponent), TEXT("Received attached component"));
+
+		if (IsValid(AttachedComponent))
+		{
+			RequireEqual_Int(AttachedComponent->ReplicatedValue, SpatialTestAttachedComponentReplicationValues::InitialValue,
```

Comment:
Nit: this should probably be `AssertEqual` since this wouldn't be something that would have a different value at first and later change to the right value.

Reply:
True - AssertEqual would print stuff on every pass though which will fill up logs. Still, agree with the change 👍